dtfs-central-api/src/services/utilisation-report-data-validator/utilisation-report-cell-validators/generate-payment-currency-error.ts

```diff
@@ -1,11 +1,18 @@
-const validator = require('validator');
-const { UTILISATION_REPORT_HEADERS } = require('@ukef/dtfs2-common');
+import validator from 'validator';
+import { UTILISATION_REPORT_HEADERS } from '@ukef/dtfs2-common';
+import { UtilisationReportRowValidationErrorGenerator } from './types/validation-error-generator';
 
-const generatePaymentCurrencyError = (csvDataRow) => {
-  if (!csvDataRow[UTILISATION_REPORT_HEADERS.PAYMENT_EXCHANGE_RATE]?.value && !csvDataRow[UTILISATION_REPORT_HEADERS.PAYMENT_CURRENCY]?.value) {
+/**
+ * Generate and return error for payment currency entry in the csv row if value is invalid
+ * @param csvDataRow - The row data for which to validate the payment currency
+ * @returns The error if the payment currency entry is invalid, null if the payment currency entry is valid
+ */
+export const generatePaymentCurrencyError: UtilisationReportRowValidationErrorGenerator = (csvDataRow) => {
+  const paymentCurrencyValue = csvDataRow[UTILISATION_REPORT_HEADERS.PAYMENT_CURRENCY]?.value;
+  if (!csvDataRow[UTILISATION_REPORT_HEADERS.PAYMENT_EXCHANGE_RATE]?.value && !paymentCurrencyValue) {
```

Comment:
could this not also have a const for `csvDataRow[UTILISATION_REPORT_HEADERS.PAYMENT_EXCHANGE_RATE]?.value` like there is for paymentCurrencyValue

Reply:
I didn't pull it out because unlike the other one it is only used in one place and the other one needing pulling out to get the type checking follow on from the if statement checking it was undefined. But I will pull it out now.